src/Repository/BookingCodes.php

```diff
@@ -75,11 +75,12 @@ public static function getCodes( int $timeframeId, int $startDate = null, int $e
 
 			//check, if we have enough codes for the timeframe or if we need to generate more
 			//we only need to check, if we have an open-ended timeframe
-			//we check, if the end date of the last generated code is before the end date of the requested time period
-			if ( ! $timeframe->getRawEndDate() && self::getLastCode( $timeframe )
-				&& strtotime( self::getLastCode( $timeframe )->getDate() ) < strtotime( $endDate )
-			) {
-				$startGenerationPeriod = new \DateTime( self::getLastCode($timeframe)->getDate() );
+			// NOTE: there used to be a check if generation is necessary by checking the date of the last code. However,
+			// as the codes are never deleted anymore, it is possible that they get fragmented with date gaps and the
+			// check is not trivial anymore. Is is easier and safer to always try to generate codes. It is no serious
```

Comment:
Das würde ich gerne nochmal checken. Das bedeutet ja dann, dass für jeden unendlichen Timeframe beim Speichern immer ein ganzes Jahr an Buchungscodes generiert wird, selbst wenn das nicht nötig wäre. Oder?

Reply:
Habe gerade mal nachgeschaut, wenn zum Beispiel bei einem Backend Aufruf nur die Codetabelle gelesen wird, wird die getCodes() ausgeführt, die dann über die generatePeriod() 365x eine SQL Anfrage stellt, ob der Code für jeden der Tage schon existiert. Vielleicht bräuchte es noch eine Möglichkeit zu prüfen wie viele codes für einen Zeitraum existieren...

Reply:
Hmm, habe das gerade mal durchprobiert aber es scheint mir doch so, dass es ja dann nur für die Randfälle in denen ein Timeframe 2x pro Tag aufgerufen wird eine Performanceerleichterung gibt. Ich würde sagen wir lassen das erstmal so und wenn sich wer beschwert ändern wir das wieder.

Reply:
> Das würde ich gerne nochmal checken. Das bedeutet ja dann, dass für jeden unendlichen Timeframe beim Speichern immer ein ganzes Jahr an Buchungscodes generiert wird, selbst wenn das nicht nötig wäre. Oder?

Du müsste stimmen, ja. Wenn nicht aus dem Cache geliefert wird, dann dürfte tatsächlich typischerweise 365 SQL-SELECT-Abfragen laufen. Das könnte man optimieren. Aber lohnt es sich? Diese Lese-Abfragen ohne irgendwelche JOINs dürften meinem Verständnis nach extrem günstig sein und wenn überhaupt nur dem Site-Admin, nicht dem Websitebesucher auffallen.